.github/workflows/rust.yml

```diff
@@ -0,0 +1,22 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build
+      run: cargo build --verbose
```

Comment:
While the `ubuntu-latest` image brings Rust tooling, it will usually be upgraded, meaning that compiler errors may switch and break often - this has happened in the past.

I ask you to add a proper versioning setup, like https://github.com/jmg-duarte/sealed-rs/blob/900cf84be5a48ac1ddb895990a77d49b656e840f/.github/workflows/ci.yml#L17-L21 
Or just a bash script that downloads a fixed version of Rust

Reply:
I adjusted the ci from seald-rs to this project.
It is using the MSRV cargo msrv determined.